### PR TITLE
!!! FEATURE: Use google api package to retrieve statistics

### PR DESCRIPTION
**Pull request by Sebobo**
This change will:

* Remove custom api connection code and use api package for this
* Replace manual authentication in the backend with authentication via key file

### CODE REVIEW

**Comment by jonnitto** [comment]
Very cool! Will test it asap

**Comment by Sebobo** [comment]
Postgres Migration is missing and I'm having trouble generating it, always getting errors just trying to connect a dockerized instance.

File: Classes/Service/Reporting.php [hunk 1]
```diff
@@ -102,7 +108,7 @@ public function getNodeStat(NodeInterface $node, ControllerContext $controllerCo
             $parameters['max-results'] = $statConfiguration['max-results'];
         }
         $gaResult = $this->analytics->data_ga->get(
-            'ga:' . $siteConfiguration->getProfileId(),
+            'ga:' . $siteConfiguration['profileId'],
```
Comment (cvette) [inline Classes/Service/Reporting.php]: Site and default settings should be merged, to have consistent behavior.

File: Documentation/index.rst [hunk 2]
```diff
@@ -23,12 +25,9 @@ Configuration requires some setup on the side of Google, before the package can
 In a nutshell:
 
 #. Log in to the Google Developers Console
-#. Set up the Consent screen
-#. Create a web application Client ID
-#. Create a public access Server key
+#. Create a service account key
```
Comment (cvette) [inline Documentation/index.rst]: You also have to add permissions for the created service account in your Google Analytics account.

File: Resources/Private/Translations/en/Main.xlf [hunk 3]
```diff
@@ -0,0 +1,25 @@
...
+    <file original="" product-name="Neos.GoogleAnalytics" source-language="en" datatype="plaintext">
+        <body>
+            <trans-unit id="accounts" xml:space="preserve">
+                <source>Authenticated accounts</source>
+            </trans-unit>
+            <trans-unit id="site.name" xml:space="preserve">
+                <source>Site</source>
+            </trans-unit>
+            <trans-unit id="site.gaProfile" xml:space="preserve">
+                <source>Google Analytics Profile</source>
```
Comment (cvette) [inline Resources/Private/Translations/en/Main.xlf]: Maybe add a screenshot that shows where to find the profile id. In the German GA UI it is not very clear what the profile actually is.

File: Resources/Private/BackendFusion/Components/FlashMessages.fusion [hunk 4]
```diff
@@ -0,0 +1,16 @@
+prototype(Neos.GoogleAnalytics:Component.FlashMessages) < prototype(Neos.Fusion:Component) {
+    flashMessages = ${[]}
+
```
Comment (jonnitto) [inline Resources/Private/BackendFusion/Components/FlashMessages.fusion]:
```suggestion
    @if.hasMessages = ${Type.isArray(this.flashMessages) && Array.length(this.flashMessages)}
```

File: Resources/Private/BackendFusion/Components/FlashMessages.fusion [hunk 5]
```diff
@@ -0,0 +1,16 @@
...
+    renderer = afx`
+        <div id="neos-notification-container" class="neos-notification-top" @if.hasMessages={props.flashMessages}>
```
Comment (jonnitto) [inline Resources/Private/BackendFusion/Components/FlashMessages.fusion]:
```suggestion
        <div id="neos-notification-container" class="neos-notification-top">
```

File: Resources/Private/BackendFusion/Components/FlashMessages.fusion [hunk 6]
```diff
@@ -0,0 +1,16 @@
...
+            <Neos.Fusion:Loop items={props.flashMessages} itemName="message">
+                <div class="neos-notification neos-notification-error">
+                    <div class="neos-notification-content">
+                        <i class="fas fa-error"></i>
+                        <div class="neos-notification-heading" @if.hasTitle={message.title}>{message.message}</div>
```
Comment (jonnitto) [inline Resources/Private/BackendFusion/Components/FlashMessages.fusion]: The title never gets shown. Is that on purpose?

Comment (Sebobo) [reply Resources/Private/BackendFusion/Components/FlashMessages.fusion]: No that was a mistake, I updated the prototype and split the rendering. Please check.

File: Resources/Private/BackendFusion/Components/FlashMessages.fusion [hunk 7]
```diff
@@ -0,0 +1,32 @@
...
+    severity.@process.replaceNoticeStatus = ${value == 'notice' ? 'info' : value}
+
+    renderer = afx`
+        <div class={'neos-notification neos-notification-' + props.severity}>
+            <div class={'neos-notification-content' + (props.message.message ? ' expandable' : '')}>
+                <i class={'fas fa-' + props.severity}></i>
+                <div class="neos-notification-heading" @if.hasTitle={props.message.title}>
+                    {props.message.title}
+                </div>
+                <div class="neos-expand-content">{props.message.message}</div>
```
Comment (jonnitto) [inline Resources/Private/BackendFusion/Components/FlashMessages.fusion]:
```suggestion
                <div class="neos-expand-content" @if.hasMessage={props.message.message}>{props.message.message}</div>
```

File: Resources/Private/BackendFusion/Components/FlashMessages.fusion [hunk 8]
```diff
@@ -0,0 +1,32 @@
...
+prototype(Neos.GoogleAnalytics:Component.FlashMessages.Message) < prototype(Neos.Fusion:Component) {
+    message = ${{}}
+
+    severity = ${String.toLowerCase(this.message.severity)}
+    severity.@process.replaceOKStatus = ${value == 'ok' ? 'success' : value}
+    severity.@process.replaceNoticeStatus = ${value == 'notice' ? 'info' : value}
+
+    renderer = afx`
+        <div class={'neos-notification neos-notification-' + props.severity}>
+            <div class={'neos-notification-content' + (props.message.message ? ' expandable' : '')}>
```
Comment (jonnitto) [inline Resources/Private/BackendFusion/Components/FlashMessages.fusion]: I would suggest writing this like that:

```suggestion
            <div class={['neos-notification-content',  props.message.message ? 'expandable' : null]}>
```

Comment (Sebobo) [reply Resources/Private/BackendFusion/Components/FlashMessages.fusion]: Cool, didn't know that trick 😄

**Pull request merged**